src/codeu/chat/server/Controller.java

```diff
@@ -189,6 +189,32 @@ public User newUser(Uuid id, String name, Time creationTime) {
     return user;
   }
 
+
+  @Override
+  public void addMember(Uuid user, Uuid conversation) {
+    ifIdFree(user) {
+      throw new IOException("Must add an existing user");
+    }
+    final User userToAdd = model.userById().first(user);
+    final ConversationPayload _conversation = model.conversationPayloadById().first(conversation);
+
+    // confused on what I should set this to
+    UserControl level();
```

Comment:
You need to create dedicated class under codeu.chat.common (UserControlList) that holds the permission info:
* user id
* conversation id
UserControl value

Then you need to update model class to hold the new permission objects.